### ISSUE TITLE
Plugin fails when extension sslconf in PHP is not enabled

### ISSUE THREAD

**Issue by fho4train** [splitbrain/dokuwiki-plugin-smtp#33]
Message of SMTP test:
_There was an unexpected problem communicating with SMTP: Could not open SMTP Port._
This happened on a Synology NAS with os DSM 6.2.2-24922 Update 3. Dokuwiki was manually installed and was supposed to run with Web Station. Resolution was to enable extension sslconf in the PHP environment of the Synology Web Station. The Dokuwiki installed by Synology has this extension enabled, but this is not the case for Synology Web station.
This extension is standard enabled by XAMPP.
The behaviour of the plugin is correct, but it took a lot of time to find the problem with this cryptic message. The request is to add a remark about this required extension in the documentation of the plugin.

**Comment by mooks42** [comment]
Hi,

I'm getting the same error...
There was an unexpected problem communicating with SMTP: Could not open SMTP Port.
SMTP log:
Set: the server
Set: the auth
Set: a message will be sent
Connecting to mail.*******.** at 25

Release 2018-04-22b "Greebo" and latest Pluginversion.
Fresh install on CentOS and also mod_sslconf installed, still the same problem.

I've also tried to send the mail without encryption with two different smtp-servers, still the same cryptic and not very helpful error message.

**Comment by splitbrain** [comment]
I don't think this is related to SSL/TLS at all. The failure happens at the TCP layer before any encryption — stream_socket_client() couldn't open a socket to the host:port at all. That almost always means the port is blocked by a firewall, the host is wrong, or DNS doesn't resolve.
